src/find_email_addresses.py

```diff
@@ -0,0 +1,63 @@
+import sys
+import argparse
+import validators 
+import requests
+import requests.exceptions
+from bs4 import BeautifulSoup 		# wow, not working with Python3.6, using Python3.5 
+import re
+from collections import deque
+email_pattern = re.compile("[\w\.-]+@[\w\.-]+")
+
+
+def findObjects(rootUrl):
+		processed_urls=set()	# unique items
+		emails=set()		# unique items
+		urls=deque([rootUrl])	# good FIFO
+		# obtain urls beyond root 
+		while len(urls):
+			urlnext=urls.popleft()
+			if not urlnext in processed_urls:
+				processed_urls.add(str(urlnext))
+				try:
+					response=requests.get(urlnext)
+					page=BeautifulSoup(response.text, 'html.parser')
+					
+					#print("Collecting emails from {}".format(urlnext))
+					emailSet=set(re.findall(email_pattern, str(page)))
+					emails=emails.union(emailSet)
+					# collect href references
+					for atag in page.find_all("a"):
```

Comment:
Did you try using a regexp for this as well, like you're doing for emails?

Reply:
No. I have little hands-on html experience, enough to insert hyperlink or image (and make text bold or strikeout), so I looked to proper terms and this tag is called anchor tag, "href" indicates a link.  A lot of resources for processing like http://web.stanford.edu/~zlotnick/TextAsData/Web_Scraping_with_Beautiful_Soup.html. Never used BeautifulSoup before, useful instrument.